gridfinity-rebuilt-baseplate.scad

```diff
@@ -91,13 +91,17 @@ module gridfinityBaseplate(gridx, gridy, length, dix, diy, sp, sm, sh) {
                 linear_extrude(10*(h_base+off), center = true)
                 profile_skeleton();
             }
+            else if (sp == 4) {
+                linear_extrude(10*(h_base+off), center = true)
+                profile_skeleton_minimal();
```

Comment:
You don't need to separate this into a separate module. You don't even need the call to `linear_extrude`. Just use `rounded_square(length-6, 10*(h_base+off), r_fo3)`. Then you don't need to refactor anything. Also, the `-6` seems a bit magic, are there any parameters that should be dependent on?

Reply:
The 6 needs to be calculated, it is not a static value. I need the same value as calculated in line 121 of gridfinity-rebuild-utility.scad. This will be the following calculation. It is going a little bit over my head to calculate the width for the bottom part of the base of a bin. There is a separate calculation needed for the `y` size. Do you by any chance see a easier way to calculate the value?

```c
gx = gridx
dx = 1
dbnxt = [for (i=[1:5]) if (abs(gx*i)%1 < 0.001 || abs(gx*i)%1 > 0.999) i];
dbnx = 1/(dx==0 ? len(dbnxt) > 0 ? dbnxt[0] : 1 : round(dx));
xx = dbnx*l-0.05
xx-2*r_c2-2*r_c1+o
````

Reply:
you are looking in the wrong place. just look at how the skeleton profile does it - `l` is the size. basically it would be `length-2*r_c2-2*r_c1` instead of `length-6`. (`r_c2` and `r_c1` are the sizes of the chamfers on the bases)